cdragontoolbox/hashes.py

```diff
@@ -667,6 +700,19 @@ def grep_wad(self, wad):
 
         with open(wad.path, 'rb') as f:
             for wadfile in wad.files:
+                if wadfile.path_hash == 2155072684501898278 or wadfile.path == 10561690728639675755: # quality coding
```

Comment:
This is currently in place as these two files contain all included files of the DATA.wad.client and the Scripts.wad.client file. These have 4 byte prefixes and are currently not checked at all by grep_wad. I should probably figure out a way of making this a more general check, as I feel like this is a kinda sloppy check.

Reply:
Until those hashes are resolved I see two possibilities:
- keep hashes hardcoded as you suggested
- use `grep_file()` as fallback for unknown/generic binary files

The second solution would require to filter formats known to not contain paths (such as .dds). Code would be something like this:
```py
if wadfile.ext in ('bin', 'inibin'):
    # bin-specific processing
elif wadfile.ext == 'preload':
    # preload-specific processing
...
elif wadfile.ext in ('dds', 'tga', ...):
    pass  # don't grep filetypes known to not contain paths
else:
    # fallback: search for path-looking strings
    self.grep_file(data=wadfile.read_data(f))
```

Then update `grep_file()` to also support data rather than only a path (better than copy-pasting the implementation):
```py
def grep_file(self, path=None, data=None):
    if path is None:
        with open(path, "rb") as f:
            data = f.read()
    elif data is None:
        raise TypeError("either path or data must be provided")
    ....
```